common/djangoapps/nodebb/signals/handlers.py

```diff
@@ -5,15 +5,56 @@
 from django.dispatch import receiver
 
 from common.lib.nodebb_client.client import NodeBBClient
+from lms.djangoapps.onboarding_survey.models import ExtendedProfile, UserInfoSurvey, InterestsSurvey
 from nodebb.models import DiscussionCommunity
-from lms.djangoapps.onboarding_survey.models import ExtendedProfile
 from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
 from student.models import ENROLL_STATUS_CHANGE, EnrollStatusChange
 from xmodule.modulestore.django import modulestore
 
 log = getLogger(__name__)
 
 
+@receiver(post_save, sender=UserInfoSurvey)
+@receiver(post_save, sender=ExtendedProfile)
+@receiver(post_save, sender=InterestsSurvey)
+def sync_user_info_with_nodebb(sender, instance, **kwargs):  # pylint: disable=unused-argument, invalid-name
+    """ Sync user information with  """
+    user = instance.user
+
+    if user:
+        try:
+            extended_profile = user.extended_profile
```

Comment:
This is not efficient. we are hitting DB at-least 3 times for each save call on any of UserInfoSurvey, ExtendedProfile or InterestsSurvey. I think we update data based on sender without grabbing the data from these models again. something like 
```
        if sender == InterestsSurvey:
               data_to_sync = {
                     'interests': ' '.join([area.label for area in interest_survey.capacity_areas.all()])
              }
         elif sender == ExtendedProfile:
             data_to_sync = {
                 "first_name": extended_profile.first_name,
                 "last_name": extended_profile.last_name,
             }
         else:
             data_to_sync = {
                 "city_of_residence": user_info_survey.city_of_residence,
                 "country_of_residence": user_info_survey.country_of_residence
             }

```